bot/exts/moderation/infraction/infractions.py

```diff
@@ -76,31 +76,52 @@ async def kick(self, ctx: Context, user: FetchedMember, *, reason: t.Optional[st
         await self.apply_kick(ctx, user, reason)
 
     @command()
-    async def ban(self, ctx: Context, user: FetchedMember, *, reason: t.Optional[str] = None) -> None:
-        """Permanently ban a user for the given reason and stop watching them with Big Brother."""
-        await self.apply_ban(ctx, user, reason)
+    async def ban(
+        self,
+        ctx: Context,
+        user: FetchedMember,
+        duration: t.Optional[Expiry] = None,
+        *,
+        reason: t.Optional[str] = None
+    ) -> None:
+        """
+        Permanently ban a user for the given reason and stop watching them with Big Brother.
+
+        If duration is specified, it temporarily bans that user for the given duration.
+        """
+        await self.apply_ban(ctx, user, reason, expires_at=duration)
 
     @command(aliases=('pban',))
     async def purgeban(
         self,
         ctx: Context,
         user: FetchedMember,
-        purge_days: t.Optional[int] = 1,
+        duration: t.Optional[Expiry] = None,
         *,
         reason: t.Optional[str] = None
     ) -> None:
         """
-        Same as ban but removes all their messages for the given number of days, default being 1.
+        Same as ban but removes all their messages of the last 24 hours.
 
-        `purge_days` can only be values between 0 and 7.
-        Anything outside these bounds are automatically adjusted to their respective limits.
+        If duration is specified, it temporarily bans that user for the given duration.
         """
-        await self.apply_ban(ctx, user, reason, max(min(purge_days, 7), 0))
+        await self.apply_ban(ctx, user, reason, 1, 0, expires_at=duration)
```

Comment:
What is that 0 for? please test changes before pushing them ^^"

Reply:
Did test the others, missed out on this one.